bb_cli/Cargo.toml

```diff
@@ -11,15 +11,15 @@ bb_common = { path = "../bb_common" }
 log = "0.4"
 
 # auth
-rand = "0.8.3"
-rsa = "0.8.1"
+rand = "0.8"
+rsa = "0.8"
 
 # networking
-mio = "0.8.5"
-crossbeam-channel = "0.5.1"
-parking_lot = "0.12.1"
+mio = "0.8"
+crossbeam-channel = "0.5"
+parking_lot = "0.12"
 
 # for the cli
-rustyline = "10.1.1"
-crossterm = "0.26.0"
-ansi_term = "0.12.1"
+rustyline = "11.0"
+crossterm = "0.26"
+ansi_term = "0.12"
```

Comment:
I'm curious why the patch versions were removed here? I initially left those in to avoid dependencies down the line causing breakage for people trying to clean build bamboo. I could be convined otherwise, but regardless I think it'd be best to be consistent across all dependencies. There are still a couple that have the patch version set, so it'd be nice if we could just pick one and use the same system throughout.

Reply:
It is recommended to exclude patch versions and only include major and minor versions. Patch versions typically do not include any breaking changes, and users should always aim to use the latest version available. Specifying the patch version should only be necessary if there are specific issues with the newest version.

Reply:
I don't particularly agree when it comes to every dependency. People make mistakes, and breaking changes often make it into patch releases regardless of best intentions. It seems like patch versions should just be left as-is, and bumped when all the other versions get bumped. I realize Cargo.lock can mitigate this, but I'd rather not break my other tools that depend on bamboo as a library.